lib/activedirectory.js

```diff
@@ -808,6 +810,16 @@ function getGroupMembershipForDN(opts, dn, stack, callback) {
   });
 }
 
+function callOnlyOnce(callback) {
+  var hasBeenCalled = false;
+  return function() {
+    if (!hasBeenCalled) {
+      hasBeenCalled = true;
+      callback.apply(this, arguments);
+    }
```

Comment:
@davidrapin Maybe we could add some logging to detect if this issue happens? I'm aware `console` isn't the "cleanest" solution, but I think it's fine in this niche use-case.

```suggestion
    if (hasBeenCalled) {
      console.trace('ignoring already called callback');
    } else {
      hasBeenCalled = true;
      callback.apply(this, arguments);
    }    
```